src/Native/System.Native/pal_networking.cpp

```diff
@@ -213,12 +216,33 @@ extern "C" int32_t IPv4StringToAddress(const uint8_t* address, uint8_t* buffer,
     return ConvertGetAddrInfoAndGetNameInfoErrorsToPal(result);
 }
 
+static void AppendScopeIfNecessary(uint8_t* string, int32_t stringLength, uint32_t scope)
```

Comment:
> I think that typing our strings as char*s, etc. is disingenuous.

Isn't this how libc and other platform libraries do it?  And they're often giving back or filling in UTF8-encoded strings.

Regardless, I see that you're using the same typing used elsewhere in this code, so for consistency with the surrounding code, it's fine.  But I think we should go back through and ensure consistency across the shims, with whatever type we decide. Personally, I'd prefer to see us stick with ```char*```.